doc/aaa/SONiC RBAC HLD.md

```diff
@@ -0,0 +1,381 @@
+
+At bootup time of the manageability framework (or gNMI container), it is recommended to cache the [User DB](#326-user-db) if not already cached. This is because every command needs to access the information in the User DB in order to access the information. Alternately, instead of caching the entire User DB, the information can be cached once the record is read from the DB. Additionally, the Translib must listen to change notifications on the User DB in order to keep its cache current.
+
+As described in section [1.1.1.4 Linux Groups](#1114-linux-groups), the enforcement of the users and roles in Linux will be done via the Linux groups. A user can use Linux commands on the Linux shell and create users and Linux groups (which represent the roles). This will mean that the information in the User DB is no longer current. In order to keep the information in the User DB and the Linux `/etc/passwd` file in sync, a service must run on the host (not the container) to keep the databases in sync.
+
+Since Translib is the main authority on authorized operations, this means that the NBIs cannot render to the user what they are and are not allowed to do. The CLI, therefore, renders the entire command tree to a user, even the commands they are not authorized to execute.
+
+#### 1.1.1.4 Linux Groups
+RBAC will be facilitated via Linux Groups:
+- Local users
+  - Local user with `Operator` role is added into `docker` group
+  - Local user with `Admin` role is added into `admin` group and is a `sudoer`.
+- Remote users
+  - Remote user with `Operator` role mapped to a global `remote-user` user who is part of `docker` group
+  - Remote user with `Admin` role mapped to a global `remote-user-su` user who is part of `admin` group and is a `sudoer`
+  - In the future, this "global user" approach will be revisited so that remote users are authenticated with their own username so that their activities may be properly audited.
+
+#### 1.1.1.5 Certificate-based Authentication for gNMI and REST
+For the initial release, it will be assumed that certificates will be managed outside of the NBIs. That is, no CLIs or REST/gNMI interfaces will be implemented to support public key infrastructure management for certificates and certificate authorities. Certificates will be manually generated and copied into the system via Linux utilities.
```

Comment:
Both routes are being taken. The REST server will accept connections from two sources:
(1) CLISH from localhost. This will use self-signed certificates.
(2) External sources. If certificate authentication is used for these external sources, either (a) their certificates need to be installed in the trust store, or (b) their CA's certificate needs to be installed in the trust store. 

I'm saying that for certificates in the second scenario, those will need to be manually managed and we're not providing a way in the first release to do so.

Reply:
I have added language to this section to clarify the difference between certs for CLISH-based connections and certs for external sources.